### DISCUSSION TITLE
Accessing FERC 714 2010 - 2020 via Docker + Jupyter

### DISCUSSION THREAD

**Discussion by kristinaavery** [catalyst-cooperative/pudl#2467] [Help Me!] [answered]
Hello! I am trying to access hourly demand data from the FERC 714 form for the years 2010 - 2020 for WECC. From this [link](https://catalystcoop-pudl.readthedocs.io/en/latest/data_sources/ferc714.html#how-much-of-the-data-is-accessible-through-pudl), my understanding is that PUDL has normalized the FERC 714 data from 2006 - 2020. I was unable to pull bulk data from the data tables using datasette, so I set up Docker + Jupyter . However, I am having a hard time finding the [demand tables](https://catalystcoop-pudl.readthedocs.io/en/latest/data_dictionaries/pudl_db.html#demand-hourly-pa-ferc714) within the pudl_db. Am I looking in the wrong place? I see the data is available from 2021 on from the xbrl_db, but I need access to historical data.

**Comment by zaneselvans** [comment, marked as answer]
Hey there, if you'd like to work with all of the hourly data, rather than just browsing it, your best bet is probably to download the full PUDL DB from the most recently successful nightly build. You can find links to the nightly build outputs [on the front page of our documentation](https://catalystcoop-pudl.readthedocs.io/en/latest/index.html#nightly-data-builds) and [the README for this repo](https://github.com/catalyst-cooperative/pudl/#nightly-data-builds).

Once you've downloade `pudl.sqlite` you can read the demand table using pandas like this:

```python
import sqlalchemy as sa
import pandas as pd

engine = sa.create_engine("sqlite:////path/to/your/downloaded/pudl.sqlite")

# The hourly time series:
dhpa_ferc714 = pd.read_sql("demand_hourly_pa_ferc714", engine)

# A table that maps respondent IDs to their names:
respondents_ferc714 = pd.read_sql("respondent_id_ferc714", engine)
```

But you should also be able to [browse the hourly demand table in Datasette](https://data.catalyst.coop/pudl/demand_hourly_pa_ferc714). Was the link from the [data dictionary](https://catalystcoop-pudl.readthedocs.io/en/latest/data_dictionaries/pudl_db.html#demand-hourly-pa-ferc714) page not working?

**Reply by kristinaavery** [reply]
Hello! I was able to browse small sections of the data in datasette, but I needed access to 10 years of data so I tried docker + jupyter instead. I had trouble finding the hourly demand data at first, but did end up finding it using the examples in the second notebook for pudl outputs. So, I got what I needed! Thank you!!
